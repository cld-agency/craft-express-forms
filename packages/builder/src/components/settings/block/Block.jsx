import React from 'react';
import PropTypes from 'prop-types';
import Info from '@xf/builder/components/common/info/Info';
import translate from '@xf/builder/functions/translator';

const Block = ({ title, description, extraBlock, children }) => (
  <div className="block">
    <h3>
      {translate(title)} <Info description={translate(description)} />
      {extraBlock}
    </h3>
    {children}
  </div>
);

Block.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  extraBlock: PropTypes.object,
};

export default Block;
