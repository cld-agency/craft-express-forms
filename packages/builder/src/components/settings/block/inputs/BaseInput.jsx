import Tooltip from '@material-ui/core/Tooltip/Tooltip';
import React from 'react';
import PropTypes from 'prop-types';
import translate from '../../../../functions/translator';

class BaseInput extends React.Component {
  static propTypes = {
    label: PropTypes.string,
    description: PropTypes.string,
    required: PropTypes.bool,
    name: PropTypes.string,
    value: PropTypes.any,
    saveHandler: PropTypes.func,
    valueToCamelCase: PropTypes.bool,
    disabled: PropTypes.bool,
  };

  static defaultProps = {
    description: null,
    required: false,
  };

  getBlockClasses() {
    return [];
  }

  renderInput() {}

  render() {
    const { label, required, description } = this.props;

    const blockClasses = ['block-row', ...this.getBlockClasses()];

    return (
      <div className={blockClasses.join(' ')}>
        {label && (
          <div className="label">
            <Tooltip title={translate(description) ?? ''} placement="bottom">
              <label className={required ? 'required' : ''}>{translate(label)}</label>
            </Tooltip>
          </div>
        )}
        <div className="input ltr">{this.renderInput()}</div>
      </div>
    );
  }
}

export default BaseInput;
