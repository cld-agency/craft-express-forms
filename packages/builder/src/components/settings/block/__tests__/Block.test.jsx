import React from 'react';
import { shallow } from 'enzyme';
import Block from '../Block';
import BaseInput from '../inputs/BaseInput';

class TestInput extends BaseInput {
  renderInput() {
    return '';
  }
}

describe('<Block />', () => {
  it('renders & displays correctly', () => {
    const wrapper = shallow(<Block title="Block Title" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays a title', () => {
    const wrapper = shallow(<Block title="Block Title" />);
    expect(wrapper.find('h3').text()).toStrictEqual('Block Title <Info />');
  });

  it('renders children', () => {
    const wrapper = shallow(
      <Block title="Block Title">
        <TestInput label="Testing one" name="test1" value="test1" description="A description" />
        <TestInput label="Testing two" name="test2" value="test2" required={true} />
      </Block>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('TestInput')).toHaveLength(2);
  });
});
