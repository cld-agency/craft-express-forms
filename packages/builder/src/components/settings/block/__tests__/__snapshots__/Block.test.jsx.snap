// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block /> renders & displays correctly 1`] = `
<div
  className="block"
>
  <h3>
    Block Title
     
    <Info />
  </h3>
</div>
`;

exports[`<Block /> renders children 1`] = `
<div
  className="block"
>
  <h3>
    Block Title
     
    <Info />
  </h3>
  <TestInput
    description="A description"
    label="Testing one"
    name="test1"
    required={false}
    value="test1"
  />
  <TestInput
    description={null}
    label="Testing two"
    name="test2"
    required={true}
    value="test2"
  />
</div>
`;
